### ISSUE TITLE
ValueError: The truth value of a DataFrame is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().

### ISSUE THREAD

**Issue by salmanyusuf** [dreamquark-ai/tabnet#336]
Hi, I am not sure while `drop_last=False` in the code below gives an error. Any help will be greatly appreciated! 

`import pytorch_tabnet
from pytorch_tabnet.tab_model import TabNetClassifier
import torch

from sklearn.preprocessing import LabelEncoder
from sklearn.metrics import roc_auc_score, accuracy_score

clf = TabNetClassifier(optimizer_fn=torch.optim.Adam,
                       optimizer_params=dict(lr=2e-2),
                       scheduler_params={"step_size":50, # how to use learning rate scheduler
                                         "gamma":0.9},
                       scheduler_fn=torch.optim.lr_scheduler.StepLR,
                       mask_type='entmax' # "sparsemax"
                      )

clf.fit(
    X_train=X_train_smote, y_train=y_train_smote,
    eval_set=[(X_train, y_train), (X_test, y_test)],
    eval_name=['train', 'test'],
    eval_metric=['auc'],
    max_epochs=1000 , patience=20,
    batch_size=1024, virtual_batch_size=128,
    num_workers=0,
    weights=1,
    drop_last=False
)`

![Screenshot 2021-10-27 at 12 20 18 AM](https://user-images.githubusercontent.com/86400044/138920162-e8ca4e40-a47c-484b-bc8c-ea0ce6740ee0.png)

**Comment by salmanyusuf** [comment]
Will do. Thank you once again!

**Comment by salmanyusuf** [comment]
Hi @Optimox, apologies for so many questions. I tried doing a randomizedsearchcv to get the best hyperparameters, however I got this error: TypeError: type object argument after ** must be a mapping, not float. I'm not really sure how to solve this. Have attached the screenshot below for reference. Thanks so much!

![Screenshot 2021-10-31 at 10 37 45 PM](https://user-images.githubusercontent.com/86400044/139588757-763eb735-ffe4-4df1-848b-100a4789bacc.png)
